updated 4.md (#2834)
webpack/webpack.js.org: src/content/migrate/4.md

---
title: To v4 from v3
group: Migrate

> sort: 3

Reviewer: Please revert, it was correct
```suggestion
sort: 1
```

contributors:

>   - [sokra]

Reviewer: Please see other md files, dont use square brackets in contributors enum
```suggestion
  - sokra
```

  - [EugeneHlushko]
  - [Kolhar730]
related:
  - title: To v2 or v3 from v1

>     url: [https://webpack.js.org/migrate/3/]

Reviewer: Please see other .md files, we don't use square brakcets for the links, they are given as a placeholder in the contributors guide
```suggestion
    url: https://webpack.js.org/migrate/3/
```

---
title: To v4 from v3

> group: Migrate

Reviewer: Yes @Kolhar730  the content looks pretty good to me, thanks a lot for taking care of this! There is one last oddity i have spotted, when you have added a group there is not a redundant header in side menu, please remove it and should be good. You can see it here in preview: https://deploy-preview-2834--webpackjsorg-netlify.netlify.com/migrate/

or on this screenshot
![Annotation 2019-07-31 212752](https://user-images.githubusercontent.com/10549495/62237742-367e4c80-b3da-11e9-9223-bd1e266f2cb8.png)

```suggestion
```

## Node.js v4

*TLDR:; Node.js versions lower than 6 are no longer supported. So, upgrade to the latest Node.js LTS verison.*

> If you are still using Node.js v4 or lower, you need to upgrade your Node.js installation to Node.js v6 or higher. Ideally you should be getting the latest Long Term Support(LTS) version of Node.js. Instructions for upgrading your Node.js version can be found [here](https://stackoverflow.com/questions/10075990/upgrading-node-js-to-latest-version). This is because the source of webpack was changed to a higher ECMAScript version in webpack 4.

Reviewer: Expanding docs on unsupported node.js version doesn't look good to me. I would suggest taking a link that you've found on stackoverflow and adding it to the previous version of this section

Author: Agreed.

## CLI

The CLI has moved to a separate package: webpack-cli. You need to install it before using webpack, see [basic setup](/guides/getting-started/#basic-setup).

> The installation guide can be found [here](/src/content/guides/installation.md).

Reviewer: Did you try to follow this link? Is it working locally?

Author: Corrected.

## Update plugins

> Many 3rd party plugins need to be updated to the latest versions to be compatible with webpack 4. Many of the popular plugins and the links to their repositories can be found [here](https://github.com/webpack-contrib/awesome-webpack#webpack-plugins).

Reviewer: ```suggestion
Many 3rd party plugins need to be updated to the latest versions to be compatible with webpack 4. Links to popular plugins can be found [here](https://github.com/webpack-contrib/awesome-webpack#webpack-plugins).
```

Author: Suggestion added.

> A more informative resource can be found [here](https://medium.com/webpack/webpack-4-migration-guide-for-plugins-loaders-20a79b927202).

Reviewer: There is a section for additional reading in our `.md` files, how about we move this link there with a title of: `webpack 4: migration guide for plugins/loaders`

Author: noted

## mode

Add the new [`mode`](/concepts/mode/) option to your config. Set it to production or development in your configuration depending on config type.

> More information on the different types of `mode`'s like `none`, `development`, and `production` are mentioned in [here](https://webpack.js.org/concepts/mode/#usage).

Reviewer: I think that this addition is redundant, previous paragraph has the same page linked

Author: Removed it.

## Node.js v4

If you are still using Node.js v4 or lower, you need to upgrade your Node.js installation to Node.js v6 or higher.
 
Instructions for upgrading your Node.js version can be found [here](https://stackoverflow.com/questions/10075990/upgrading-node-js-to-latest-version). 

## CLI

The CLI has moved to a separate package: webpack-cli. You need to install it before using webpack, see [basic setup](/guides/getting-started/#basic-setup).

The installation guide can be found [here](/guides/installation).

## Update plugins

> Many 3rd party plugins need to be updated to the latest versions to be compatible with webpack 4. Links to popular plugins can be found [here](https://github.com/webpack-contrib/awesome-webpack#webpack-plugins).

Reviewer 2: IMO the current version is fine and yes, the link part is fine.

}
```

> Alternatively you can pass it via CLI: `--mode production`/ `--mode development`

Reviewer: this is the first thing it says on the mode page we are linking in the beggining of this section, for the sake of duplication prevention, i suggest to remove this line

Author: corrected.

> T> Since the goals of the `development` and `production` versions are different. You could use `webpack-merge` to use the best of both modes. Read more [here](https://github.com/webpack/webpack.js.org/blob/master/src/content/guides/production.md)

Reviewer: I dont think that this is something to be added to a migration guide. We already have webpack-merge mentioned in the Production guide too, maybe there  is a better place to talk about it? By the way did that link work for you locally?

Author: Maybe it could serve as heads up, that a certain feature is available. But I can remove if you think it doesn't need to be there. The link works but I've corrected it now.

}
```

> T> Since the goals of the `development` and `production` versions are different. You could use `webpack-merge` to use the best of both modes. Read more [here](/guides/production)

Reviewer: I have missed this, but google accessibility test recommends us to avoid words like `here`, `see here`, `follow link` etc. Please change this:
```suggestion
T> The purpose of `development` mode and `production` mode is different. You can use `webpack-merge` as shown in [production guide](/guides/production/#setup) to optimize configurations.
```

See documentation of the [`optimization.splitChunks`](/configuration/optimization/#optimization-splitchunks) for more details. The default configuration may already suit your needs.

> More information on this change is mentioned in [here](https://gist.github.com/sokra/1522d586b8e5c0f5072d7565c2bee693).

Reviewer: Another one for additional reading section i would say

Author: noted.

## module.loaders

`module.loaders` were deprecated since webpack 2 and are now removed in favor of [`module.rules`](/configuration/module/#rule).

> ## Additional Reading

Reviewer: Put this into the file header header in related section: https://webpack.js.org/contribute/writers-guide#yaml-frontmatter

Reviewer: @Kolhar730 this is still a request, please put into related section, e.g. 
```
---
title: My Article
group: My Sub-Section
sort: 3
contributors:
  - [github username]
related:
  - title: Title of Related Article
    url: [url of related article]
---
```

## module.loaders

`module.loaders` were deprecated since webpack 2 and are now removed in favor of [`module.rules`](/configuration/module/#rule).

> ## Additional Reading

Reviewer: please move both of these links to the head of the file under line 9 in `related:` section.
